src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitStringAnalyzer.cs

```diff
@@ -0,0 +1,122 @@
+            context.RegisterSyntaxNodeAction(AnalyzeSymbol, SyntaxKind.InvocationExpression);
+        }
+
+        private static void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
+        {
+            InvocationExpressionSyntax invocationExpression = (InvocationExpressionSyntax)context.Node;
+
+            if (!(invocationExpression.Expression is MemberAccessExpressionSyntax memberAccessExpression))
+            {
+                return;
+            }
+
+            // Verify that the current call is to StringBuilder.Append
+            if (memberAccessExpression.Name.ToString() != nameof(StringBuilder.Append))
+            {
+                return;
+            }
+
+            // Check that we've called Append on an instance of StringBuilder
+            ISymbol memberSymbol = context.SemanticModel.GetSymbolInfo(memberAccessExpression).Symbol;
+            if (memberSymbol == null)
+            {
+                return;
+            }
+
+            string memberSymbolContainingTypeString = memberSymbol.ContainingType.ToString();
+            if (memberSymbol.ContainingType.ToString() != "System.Text.StringBuilder")
+            {
+                return;
+            }
+
+            ArgumentListSyntax argumentList = invocationExpression.ArgumentList;
+            SeparatedSyntaxList<ArgumentSyntax> arguments = argumentList.Arguments;
+            if (arguments.Count < 1)
+            {
+                return;
+            }
+
+            ArgumentSyntax firstArgument = arguments.First();
+            ExpressionSyntax expr = firstArgument.Expression;
+            SymbolInfo exprSymbolInfo = context.SemanticModel.GetSymbolInfo(expr);
+            ISymbol exprSymbol = exprSymbolInfo.Symbol;
+            if (exprSymbol == null)
+            {
+                return;
+            }
+
+            ImmutableArray<SyntaxReference> syntaxReferences = exprSymbol.DeclaringSyntaxReferences;
+            if (syntaxReferences.IsEmpty)
+            {
+                return;
+            }
+
+            SyntaxReference declaration = syntaxReferences.First();
+            if (!(declaration.GetSyntax() is VariableDeclaratorSyntax variableDeclarator))
+            {
+                return;
+            }
+
+            if (!(variableDeclarator.Parent is VariableDeclarationSyntax localDeclaration))
+            {
+                return;
+            }
+
+            SeparatedSyntaxList<VariableDeclaratorSyntax> variables = localDeclaration.Variables;
+            foreach (VariableDeclaratorSyntax variable in variables)
+            {
+                Location declarationLocation = variable.GetLocation();
+
+                Optional<object> constant = context.SemanticModel.GetConstantValue(expr);
+                if (constant.HasValue)
+                {
+                    object value = constant.Value;
+                    if (value is string stringValue && stringValue.Length == 1)
```

Comment:
Should it flag append of an empty string as well?

Reply:
I don't think so. `Empty` is a field in the `string` class. 1) We're not flagging other instances where we have const string fields in classes and 2) Since it's not a `const` var, I'm not sure it's value is available at compile time?

Reply:
I guess it would be pretty rare to write `builder.Append("")` and it would have negligible impact anyway.

Reply:
Actually, `builder.Append("")` would get tagged and fixed to builder.Append('') (I think there was an instance of this in dotnet/runtime somewhere). I thought you were talking about builder.Append(string.Empty), which I have not seen in our repos yet :)